toybox-0.8.13: new package

IceCodeNew opened the pull request:
Fixes: #50569

Related:

### Pre-review Checklist

<!--
This checklist is mostly useful as a reminder of small things that can easily be
forgotten – it is meant as a helpful tool rather than hoops to jump through.

At the moment of this PR you have the most information on what all the change
will affect, so please take the time to jot it down.

Put an `x` in all the items that apply, make notes next to any that haven't been
addressed, and remove any items that are not relevant to this PR.

-->

#### For new package PRs only
<!-- remove if unrelated -->
- [x] This PR is marked as fixing a pre-existing package request bug
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

IceCodeNew committed:
Add toybox package

kranurag7 commented on toybox.yaml:
```diff
@@ -0,0 +1,40 @@
+      - wolfi-base
+
+pipeline:
+  - uses: fetch
+    with:
+      uri: https://github.com/landley/toybox/archive/refs/tags/${{package.version}}.tar.gz
+      expected-sha512: b6984d475b253ecf2b2fdcb3107fab690f794442dd7e3cd8b9b598381e19eaaa0ab2c2d96f4f97b27b668c5c45e61d3bab5647015fa66314d8e565337f54d775
```
we can use git-checkout pipeline in here?

IceCodeNew replied:
Working on this, thanks for the advice

IceCodeNew committed:
switch to the git-checkout pipeline

Signed-off-by: IceCodeNew <32576256+IceCodeNew@users.noreply.github.com>

kranurag7 commented on toybox.yaml:
```diff
@@ -0,0 +1,42 @@
+  epoch: 0
+  description: "A BSD-licensed alternative to busybox"
+  copyright:
+    - license: BSD-0-Clause
```
Is this correct and spdx organized license? https://github.com/landley/toybox?tab=0BSD-1-ov-file#readme 

also is this license OSI approved?

IceCodeNew replied:
> spdx organized license

https://spdx.org/licenses/0BSD.html

Yes, it is. I just need to change the `BSD-0-Clause` into a proper short identifier `0BSD`

IceCodeNew committed:
chore: license identifier
Signed-off-by: IceCodeNew <32576256+IceCodeNew@users.noreply.github.com>

IceCodeNew committed:
fix: build fail

by matching the APKBUILD configuration of toybox

IceCodeNew commented:
I should do a local test of compilation through `melange build`, pls hold on.

kranurag7 commented:
@IceCodeNew thank you for the patch. take your time, please let me know if you've any questions. :)

kranurag7 commented:
> I should do a local test of compilation through `melange build`, pls hold on.

make package/toybox should abstract things for you.

IceCodeNew committed:
fix: the old config2help cause segfault,

which is replaced in the developing branch.

IceCodeNew committed:
Update toybox version and pipeline tag

Signed-off-by: IceCodeNew <32576256+IceCodeNew@users.noreply.github.com>

IceCodeNew committed:
Temporary fix: disable `make test`

The upstream fixed tests after releasing 0.8.13, and has not released a new tag yet.
The disabled tests can be re-enabled after the upstream releases a new version

Signed-off-by: IceCodeNew <32576256+IceCodeNew@users.noreply.github.com>

IceCodeNew committed:
fix: make tests is the right command to call

Signed-off-by: IceCodeNew <32576256+IceCodeNew@users.noreply.github.com>

cmwilson21 commented:
Thanks for the contribution!

We’re closing this as part of a repo-wide reset to support upcoming changes in how contributions are validated and merged. You can find more context in our discussion post here: [Upcoming changes to wolfi-dev/os](https://github.com/orgs/wolfi-dev/discussions/77550).

If you’d like to continue with this work, please feel free to reopen or resubmit your issue or PR after reviewing the updated [Contribution Guidelines](https://github.com/wolfi-dev/os/blob/main/CONTRIBUTING.md).

We appreciate your understanding and your continued contributions to Wolfi.
